llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -176,14 +176,14 @@ static uint64_t getMinimalExtentFrom(const Value &V,
   // If queried with a precise location size, we assume that location size to be
   // accessed, thus valid.
   if (LocSize.isPrecise())
-    DerefBytes = std::max(DerefBytes, LocSize.getValue());
-  return DerefBytes;
+    DerefBytes = std::max(DerefBytes, LocSize.getValue().getKnownMinValue());
+  return LocationSize(DerefBytes);
```

Comment:
...make this return `TypeSize` as well, things will be slightly cleaner.

Reply:
I think `std::max` is not happy when any of the comparison made is with scalable vectors, because it fails with
`LLVM ERROR: Invalid size request on a scalable vector` at `std::max` . Should this be marked as a TODO?

Reply:
That sounds like you forget the getKnownMinValue() call?

Reply:
so at the end, should I make it return `TypeSize::getFixed(DerefBytes)` instead of `TypeSize::get(DerefBytes, LocSize.isScalable())` ?

Because at first I was changing even `DerefBytes` to be of `TypeSize`

Reply:
Yes, what I had in mind is to just use `TypeSize::getFixed(DerefBytes)` instead of `LocationSize(DerefBytes)`, nothing more. More precisely handling scalable vectors can be a followup.